### PR TITLE
Fix heapster deployment

### PR DESCRIPTION
**Pull request by malekpour**
Remove deprecated parameters from heapster deployments that prevent pods from starting successfully.

More at: https://github.com/kubernetes/kubernetes/commit/63ccedcfa7a34ba4bed8b697a07eab9e1e14896d

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1174)

### CODE REVIEW

File: pkg/acsengine/const.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ var KubeImages = map[string]map[string]string{
 		"dashboard":       "kubernetes-dashboard-amd64:v1.6.1",
 		"exechealthz":     "exechealthz-amd64:1.2",
 		"addonresizer":    "addon-resizer:1.7",
-		"heapster":        "heapster:v1.4.0",
+		"heapster":        "heapster-arm64:v1.4.0",
```
Comment (SaaldjorMike) [inline pkg/acsengine/const.go]: arm64?

Comment (malekpour) [inline pkg/acsengine/const.go]: Just to be aligned with other images. Also kubernetes started using arm64. https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/google/heapster-controller.yaml#L48

Comment (SaaldjorMike) [inline pkg/acsengine/const.go]: you should you dont mean amd64 instead of arm64?

Comment (malekpour) [inline pkg/acsengine/const.go]: oops, typo. just fixed it.

**Comment by lachie83** [comment]
@malekpour This was updated as part of #1076

File: parts/kubernetesmasteraddons-heapster-deployment.yaml [hunk 2]
```diff
@@ -40,11 +40,9 @@ spec:
         - "--extra-cpu=0.5m"
         - "--memory=140Mi"
         - "--extra-memory=4Mi"
-        - "--threshold=5"
```
Comment (lachie83) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: We rolled the addon-resizer back across the board currently so this change is irrelevant -- #1076

File: pkg/acsengine/const.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ var KubeImages = map[string]map[string]string{
 		"dashboard":       "kubernetes-dashboard-amd64:v1.6.1",
 		"exechealthz":     "exechealthz-amd64:1.2",
 		"addonresizer":    "addon-resizer:1.7",
-		"heapster":        "heapster:v1.4.0",
+		"heapster":        "heapster-amd64:v1.4.0",
```
Comment (lachie83) [inline pkg/acsengine/const.go]: These appear to be the same image but I'm happy to keep the naming consistent. Can you please change on all released versions?

Comment (malekpour) [inline pkg/acsengine/const.go]: Sure, I did for `heapster-amd64:v1.4.0` and `heapster-amd64:v1.3.0` but left `heapster:v1.2.0` unchanged since there is no amd64 available. This is also the kubernetes patters.

File: parts/kubernetesmasteraddons-heapster-deployment.yaml (hunk 2 above)
Comment (malekpour) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: I just saw https://github.com/Azure/acs-engine/pull/1076. I still think fixing `acs-engine` to work with `addon-resizer:v2.0` is better solution compared to keep using `addon-resizer:v1.7`

If we are using `kubernetes:1.7` and `heapster:v1.4.0`, it makes sense to use `addon-resizer:v2.0`.

https://github.com/kubernetes/kubernetes/commit/63ccedcfa7a34ba4bed8b697a07eab9e1e14896d

Comment (lachie83) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: Okay. Let's split the resizer change out into a different PR and make this PR simply updating the container image

Comment (lachie83) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: Also can you let me know the changelog of features between 1.7 and 2.0 for the resizer?

Comment (malekpour) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: It seems it has rewritten entirely and references changed from 1.7 to 2.0 everywhere https://github.com/kubernetes/autoscaler/commit/93942efb7530a7d266dd7d7a0c83b0417b049471

**Comment by lachie83** [comment]
LGTM

File: pkg/acsengine/azureconst.go [hunk 4]
```diff
@@ -2,7 +2,7 @@ package acsengine
 
 import "fmt"
 
-// AUTOGENERATED FILE - last generated 2017-05-19 20:39:35
+// AUTOGENERATED FILE - last generated 2017-08-04 15:59:31
```
Comment (lachie83) [inline pkg/acsengine/azureconst.go]: This should be in a separate PR

Comment (malekpour) [inline pkg/acsengine/azureconst.go]: I agree and didn't mean to add this. I thought the other one is merged and just pushed this to my forked repo.
